SIMSBIOHUB-629: Paginate Sampling Sites & Separate Spatial Data in API Responses

mauberti-bc opened the pull request:
## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-629

## Description of Changes

- Adds a ```/sample-site/spatial``` endpoint for retrieving the geojson data for all sites in the survey
- Removes geojson property from the response when requesting all sampling sites in the survey, replacing it with a "geometry_type" string
- Paginates sampling sites on the frontend (backend pagination was already in place)
- For the sampling site accordions on the observations page, does not load accordion content until expanded
- When retrieving observations, sampling sites related to the requested observations are included in the supplementaryData object
- Removes sample site data loader from the survey context

## Testing Notes

- Should not be any double requests for sampling site data
- Since sampling periods depend on the same request as sampling methods, we still need to solve sample period pagination (records shown in the sample period table are limited to those retrieved in the latest paginated sample method request).

mauberti-bc committed:
separate spatial and non-spatial sampling site endpoints

mauberti-bc committed:
wip: add sampling info as supplementary observations data

mauberti-bc committed:
wip: add tests

mauberti-bc committed:
fix missing dependency arrays in useeffects

mauberti-bc committed:
wip: add pagination to sampling sites tables

mauberti-bc committed:
wip: pagination changes

mauberti-bc committed:
misc cleanup

mauberti-bc committed:
wip

mauberti-bc committed:
clean up jsdocs

mauberti-bc committed:
Merge branch 'dev' into SIMSBIOHUB-629

mauberti-bc committed:
fix types

NickPhura commented:
Just going to change this to a draft temporarily as we are hitting the limit on our openshift resources.

mauberti-bc committed:
Merge branch 'dev' of github.com:bcgov/biohubbc into SIMSBIOHUB-629

mauberti-bc committed:
address sonarcloud issues

NickPhura committed:
WIP changes

NickPhura committed:
Add new sampling data grid cells to observations table

NickPhura committed:
Update sampling grid controls to reset one-another, and pre-populate dropdowns

NickPhura committed:
Tweaks/updates

NickPhura commented:
@mauberti-bc 
Okay, other than any minor CSS tweaks, etc, you might have, it should be good to go.

I'll do the other TODOs in a separate PR:
- Add pagination to the period table on the survey page.
- See if I can make the dropdown modal wider (somehow its working for the taxon/site control, but not the technique/period, even though they use the same autocomplete under the hood. But havent poked at it too hard yet)

NickPhura committed:
Fix unit tests

MacDeluca commented on api/src/openapi/schemas/observation.ts:
```diff
@@ -412,7 +414,111 @@ export const observervationsWithSubcountDataSchema: OpenAPIV3.SchemaObject = {
-        }
+        },
+        sample_sites: SampleLocationSchema
+        // sample_sites: {
```
👁️

NickPhura replied:
Removed

MacDeluca committed:
Merge branch 'dev' into SIMSBIOHUB-629

NickPhura committed:
Code review updates:
- Remove commented out openapi
- Fix sample site left nav pagination
- Add placeholder text to taxon/site search cell
- Fix loading animation not stopping when selecting or cancelling search
- Fix taxon/site cell adding the selection to the list (and creating duplicate entries in the dropdown)

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-629

MacDeluca reviewed (approved):
Poked around the UI and it all seems happy
🦖
